bindings/pyroot/pythonizations/python/ROOT/__init__.py

```diff
@@ -60,7 +60,63 @@
 import sys
 from ._facade import ROOTFacade
 
-sys.modules[__name__] = ROOTFacade(sys.modules[__name__], _is_ipython)
+_root_facade = ROOTFacade(sys.modules[__name__], _is_ipython)
+sys.modules[__name__] = _root_facade
+
+# Configure meta-path finder for ROOT namespaces
+import importlib.abc
+from importlib.util import spec_from_loader
+
+
+class _RootNamespaceLoader(importlib.abc.Loader):
+    def __init__(self):
+        self._module_cache = {}
+
+    def is_package(self, fullname):
```

Comment:
This function sounds like a simple check, but it is actually the function that creates the ROOT module(s) and writes them to the cache. I think the two parts of the logic should be separate.

Reply:
I have simplified this by not using the caching. Implementing a cache for these recursive lookups (that should be pretty fast) actually looks like a premature optimization to me.